src/main/java/io/cdap/plugin/gcp/gcs/sink/DelegatingGCSOutputCommitter.java

```diff
@@ -73,39 +92,45 @@ public void setupTask(TaskAttemptContext taskAttemptContext) throws IOException
 
   @Override
   public boolean needsTaskCommit(TaskAttemptContext taskAttemptContext) throws IOException {
-    if (committerMap.isEmpty()) {
-      return false;
-    }
-
-    boolean needsTaskCommit = true;
-
-    for (OutputCommitter committer : committerMap.values()) {
-      needsTaskCommit = needsTaskCommit && committer.needsTaskCommit(taskAttemptContext);
-    }
-
-    return needsTaskCommit;
+    return true;
   }
 
   @Override
   public void commitTask(TaskAttemptContext taskAttemptContext) throws IOException {
-    for (OutputCommitter committer : committerMap.values()) {
+    for (String output : getOutputPaths(taskAttemptContext)) {
+      taskAttemptContext.getConfiguration().set(FileOutputFormat.OUTDIR, output);
```

Comment:
This needs to be done otherwise we are getting only _SUCCESS file in the final output directory after commitJob.
As this can be called from close method also, So I was thinking of leaving it as it is with committerMap.

Reply:
Can you elaborate? I don't see OUTDIR used in the FileOutputCommiter, are there any?

Reply:
Removed. It was not getting used.